### PR TITLE
Bluetooth: EDTT: Handle unknown response

### LINKED ISSUE #22085: HCI/CCO/BV-07-C & HCI/GEV/BV-01-C failing in EDTT
**Describe the bug**
HCI/CCO/BV-07-C [BR/EDR Commands Not Supported on LE Device]
&
HCI/GEV/BV-01-C [Status return for Unsupported Commands]
are failing in EDTT, due to a mismatch in the expected response from the controller to a inquiry command.
The test expects a "Command Status" event while the controller returns a “command completed” event with invalid command status.

A proposal to change the test was sent to the EDTT to align it with the controller https://github.com/EDTTool/EDTT/pull/2 , but no immediate agreement was reached about which should be the appropriate response from the controller.

**To Reproduce**
See `tests/bluetooth/bsim_bt/edtt_ble_test_app/README.txt`
And either run all tests or just 
`tests/bluetooth/bsim_bt/edtt_ble_test_app/tests_scripts/hci.sh`

**Expected behavior**
Passing tests

**Impact**
An immediate workaround is to disable the test.
But there is a significant probability that the implementation is not correct. Therefore it would be much better to clarify what is the expected behavior and either fix the controller or test.

**Environment (please complete the following information):**
 - OS: Linux (Ubuntu 18.04)
 - Toolchain: host gcc
 - Commit SHA: Today's master, but this behavior has been the same for very long.

### PR DESCRIPTION
**Pull request by joerchan**
Related to: https://github.com/EDTTool/EDTT/pull/11
I'm not sure how to handle the west manifest with the zephyr mirror.

Update HCI test application to give the same unsupported command RSP from the command complete and command status HCI events.

Enabled the tests that now are able to pass.

Remove deprecated defines and added some debug tracing that I had been using.

Fixes: #22085

### CODE REVIEW

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c [hunk 1]
```diff
@@ -129,12 +138,12 @@ static int send_hci_command(u16_t opcode, u8_t param_len, u16_t response)
 		}
 		err = bt_send(buf);
 		if (err) {
-			printk("Failed to send HCI command %d (err %d)\n",
+			LOG_ERR("Failed to send HCI command %d (err %d)\n",
```
Comment (jhedberg) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c]: minor: log subsystem doesn't require `\n` at the end

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c [hunk 2]
```diff
@@ -129,12 +138,12 @@ static int send_hci_command(u16_t opcode, u8_t param_len, u16_t response)
...
 				opcode, err);
 			error_response(err);
 		}
 	} else {
-		printk("Failed to create buffer for HCI command %u\n", opcode);
+		LOG_ERR("Failed to create buffer for HCI command %u\n", opcode);
```
Comment (jhedberg) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c]: minor: log subsystem doesn't require `\n` at the end

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c [hunk 3]
```diff
@@ -197,10 +213,17 @@ static void command_status(struct net_buf *buf)
...
+
 		edtt_write((u8_t *)&response, sizeof(response), EDTTT_BLOCK);
 		edtt_write((u8_t *)&size, sizeof(size), EDTTT_BLOCK);
 		edtt_write((u8_t *)buf->data, buf->len, EDTTT_BLOCK);
 		waiting_opcode = 0;
+	} else {
+		LOG_WRN("Not waiting for 0x(%02x) command status,"
+			" expected 0x(%02x)",
+			opcode,
+			waiting_opcode);
```
Comment (jhedberg) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c]: I think these fit on the same line

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c [hunk 4]
```diff
@@ -479,12 +503,12 @@ static void le_data_write(u16_t size)
 			}
 			err = bt_send(buf);
 			if (err) {
-				printk("Failed to send ACL Data (err %d)\n",
+				LOG_ERR("Failed to send ACL Data (err %d)\n",
```
Comment (jhedberg) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c]: minor: log subsystem doesn't require `\n` at the end

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c [hunk 5]
```diff
@@ -479,12 +503,12 @@ static void le_data_write(u16_t size)
...
 					err);
 			}
 		} else {
 			err = -2; /* Failed to allocate data buffer */
-			printk("Failed to create buffer for ACL Data.\n");
+			LOG_ERR("Failed to create buffer for ACL Data.\n");
```
Comment (jhedberg) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c]: minor: log subsystem doesn't require `\n` at the end

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c [hunk 6]
```diff
@@ -520,7 +544,7 @@ void main(void)
 	 */
 	err = bt_enable_raw(&rx_queue);
 	if (err) {
-		printk("Bluetooth initialization failed (err %d)\n", err);
+		LOG_ERR("Bluetooth initialization failed (err %d)\n", err);
```
Comment (jhedberg) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/src/main.c]: minor: log subsystem doesn't require `\n` at the end

**Review by jhedberg — CHANGES_REQUESTED** [review]
Looks generally fine to me. A couple of small things I spotted.

**Comment by aescolar** [comment]
Added DNM label due to the manifest pointing to a EDTT PR (feel free to remove whenever it is pointing back to a normal commit)

**Review by heri-developer — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**